### PR TITLE
EG-675, EG changes to support manage UI rework

### PR DESCRIPTION
**Pull request by cmrhmrc**

### CODE REVIEW

File: app/services/ApprovalService.scala [hunk 1]
```diff
@@ -53,7 +56,21 @@ class ApprovalService @Inject() (repository: ApprovalRepository, reviewRepositor
         saveProcess(createApprovalProcess(process.meta.id, process.meta.title, initialStatus, jsonProcess)) flatMap {
           case Right(savedId) =>
             repository.getById(savedId) flatMap {
-              case Right(approvalProcess) => saveReview(createApprovalProcessReview(process, reviewType, approvalProcess.version))
+              case Right(approvalProcess) =>
+                pageBuilder.pages(process) match {
+                  case Right(pages) =>
+                    saveReview(ApprovalProcessReview(
+                                UUID.randomUUID(),
+                                process.meta.id,
```
Comment (woollett-r) [inline app/services/ApprovalService.scala:64]: IntelliJ reports trailing whitespace.

File: app/services/ApprovalService.scala [hunk 2]
```diff
@@ -53,7 +56,21 @@ class ApprovalService @Inject() (repository: ApprovalRepository, reviewRepositor
...
+                    saveReview(ApprovalProcessReview(
+                                UUID.randomUUID(),
+                                process.meta.id, 
+                                approvalProcess.version,
+                                reviewType,
+                                process.meta.title,
+                                pageBuilder.fromPageDetails(pages)(ApprovalProcessPageReview(_,_,_))
+                              ))
+                  case _ =>
+                    Logger.error("Could not parse pages")
```
Comment (woollett-r) [inline app/services/ApprovalService.scala:71]: I would provide details of process: id and version.

File: app/services/ApprovalService.scala [hunk 3]
```diff
@@ -66,7 +83,7 @@ class ApprovalService @Inject() (repository: ApprovalRepository, reviewRepositor
   }
 
   def getById(id: String): Future[RequestOutcome[JsObject]] = {
-
+    println(s"getById $id")
```
Comment (woollett-r) [inline app/services/ApprovalService.scala:86]: I would remove this.

File: app/services/PageBuilder.scala [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+        case Left(err) => Left(err)
+      }
+
+    collectStanzas(key, Nil, Nil) match {
+      case Right((ks, next, linked)) =>
+        ks.head.stanza match {
+          case p: PageStanza if p.url.isEmpty || p.url.equals("/") => Left(PageUrlEmptyOrInvalid(ks.head.key))
+          case p: PageStanza =>
+            // Removed Bullet point intruction group building, TODO move to UiBuilder
```
Comment (woollett-r) [inline app/services/PageBuilder.scala:60]: As we are only interested in CalloutStanzas and QuestionStanzas to extract the page titles I don't see any need to define instruction groups in this service.

File: app/utils/ProcessUtils.scala [hunk 5]
```diff
@@ -16,53 +16,28 @@
...
       case JsSuccess(process, _) =>
         Right(process)
       case JsError(errors) =>
         logger.error(s"Parsing process failed with the following error(s): $errors")
         Left(Errors(BadRequestError))
     }
-  }
 
-  def createApprovalProcess(id: String, title: String, status: String, jsonProcess: JsObject): ApprovalProcess = {
+  def createApprovalProcess(id: String, title: String, status: String, jsonProcess: JsObject): ApprovalProcess =
```
Comment (woollett-r) [inline app/utils/ProcessUtils.scala:40]: As there is currently only one usage of this method does it really need to be here?

File: test/services/ApprovalServiceSpec.scala [hunk 6]
```diff
@@ -37,7 +37,7 @@ class ApprovalServiceSpec extends UnitSpec with MockFactory {
```
Comment (woollett-r) [inline test/services/ApprovalServiceSpec.scala:37]: I think this class is a little on the light side. As the main test process is actually empty the page builder can't be doing anything. Perhaps an update for a later day.

File: app/services/ApprovalService.scala (hunk 1 above)
Comment (cmrhmrc) [reply app/services/ApprovalService.scala:64]: will do

File: app/services/ApprovalService.scala (hunk 2 above)
Comment (cmrhmrc) [reply app/services/ApprovalService.scala:71]: oops

File: app/services/ApprovalService.scala (hunk 3 above)
Comment (cmrhmrc) [reply app/services/ApprovalService.scala:86]: :)

File: app/services/PageBuilder.scala (hunk 4 above)
Comment (cmrhmrc) [reply app/services/PageBuilder.scala:60]: yes will remove

File: app/utils/ProcessUtils.scala (hunk 5 above)
Comment (cmrhmrc) [reply app/utils/ProcessUtils.scala:40]: agree

File: test/services/ApprovalServiceSpec.scala (hunk 6 above)
Comment (cmrhmrc) [reply test/services/ApprovalServiceSpec.scala:37]: yes

File: app/services/ApprovalService.scala (hunk 2 above)
Comment (cmrhmrc) [reply app/services/ApprovalService.scala:71]: will update

**Pull request merged**